### PR TITLE
Added backup/restore functionality and sanitized registrationid

### PR DESCRIPTION
**Pull request by logan-reynolds**

### CODE REVIEW

**Comment by garemoko** [comment]
Thanks for this. I'm going to leave this open for a little bit for people to test the PR. I've got a batch of work on this plugin on my to-do list so will merge this and #115 when that work is complete and then test the three sets of changes together before making a new release.

**Comment by garemoko** [comment]
@logan-reynolds some of the new files are failing the automatic testing. Please can you review the errors here and address them: https://travis-ci.org/garemoko/moodle-mod_tincanlaunch/builds/184809634

File: launch.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
 $event->trigger();
 
 // Get the registration id.
-$registrationid = $_GET["launchform_registration"];
+$registrationid = required_param('launchform_registration', PARAM_TEXT);
```
Comment (garemoko) [inline launch.php]: @logan-reynolds please can I get you to remove this from this PR and submit it separately as it is an unrelated issue? In that separate PR, please can you explain in a little more detail the rationale (perhaps provide a link to the relevant Moodle developer documentation). 

Thanks!

**Comment by ghost** [comment]
Hello,
  @garemoko My apologies; I've overwritten this wip-MAE-2512 branch to include only work related to backup/restore.  I've submitted a new pull request with explanation in the commit comment at https://github.com/garemoko/moodle-mod_tincanlaunch/pull/123 based on my branch wip-sanitization, so that minor change can be handled separately.  Currently all TravisCI jobs seem to be failing on the PHPCPD (copy-paste detector), but without further output I'm unable to see precisely what's causing the check to fail, since this is the standard call to required_param().  My guess is that TravisCI is likely over-checking or checking details that are irrelevant in the source code, since it does seem to fix the functional problem that was causing the plugin to fail persisting across course restore previously.

Please let me know if there is any further information or help I can provide, or if there are any revisions required in the pull request I've submitted.

**Comment by garemoko** [comment]
@logan-reynolds thanks. 

I'm not sure about the copy paste detector. Looking at the error here: https://travis-ci.org/garemoko/moodle-mod_tincanlaunch/jobs/185531838 the issues seem to be with comments not being properly formatted. E.g. [This line](https://github.com/garemoko/moodle-mod_tincanlaunch/pull/122/files#diff-45f771c34694bc4333009b3372d157fdR27) is missing a full stop at the end of the comment.

**Comment by garemoko** [comment]
Merging. I'll go ahead and fix the comments before releasing.

**Pull request merged**